tests/parametric/conftest.py

```diff
@@ -376,21 +376,61 @@ def wait_for_telemetry_event(self, event_name: str, *, clear: bool = False, wait
                 pass
             else:
                 for event in events:
-                    if event["request_type"] == "message-batch":
-                        for message in event["payload"]:
-                            if message["request_type"] == event_name:
-                                if message.get("application", {}).get("language_version") != "SIDECAR":
-                                    if clear:
-                                        self.clear()
-                                    return message
-                    elif event["request_type"] == event_name:
-                        if event.get("application", {}).get("language_version") != "SIDECAR":
-                            if clear:
-                                self.clear()
-                            return event
+                    e = self._get_telemetry_event(event, event_name)
+                    if e:
+                        if clear:
+                            self.clear()
+                        return e
             time.sleep(0.01)
         raise AssertionError(f"Telemetry event {event_name} not found")
 
+    def wait_for_telemetry_configurations(self, *, clear: bool = False) -> dict[str, str]:
+        """Waits for and returns the latest configurations captured in telemetry events.
+
+        Telemetry events can be found in `app-started` or `app-client-configuration-change` events.
+        The function ensures that at least one telemetry event is captured before processing.
+        """
+        events = []
+        configurations = {}
+        # Allow time for telemetry events to be captured
+        time.sleep(1)
```

Comment:
Is this the upper bound for all tracers?

Reply:
No, it's an arbitrary value. I believe all weblog tests are run with telemetry heartbeat interval set to 0.2. So we should get at least one flush in 1 second